### ISSUE TITLE
Can not delete file.  Retry limit exceeded errors

### ISSUE THREAD

**Issue by brucejohnson** [orientechnologies/orientdb#3546]
I'm running orientdb 1.7.10 (within baasbox www.baasbox.com) on RedHat Enterprise Linux 6, with Java 1.8.0.31.  I'm loading a database via the baasbox curl interface (via urllib2 in Jython) and periodically getting errors about not being able to delete files.  Two stack traces for two different locations in the code are listed below.  I also get a similar error to the first one when I try to restart the database.  As baasbox currently uses 1.7.10 I can't upgrade oriented to 2.0.x

Any suggestions on what is causing this problem.
Thanks,
Bruce

##################################
Caused by: java.io.IOException: Can not delete file. Retry limit exceeded. (11).
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog$LogSegment.delete(OWriteAheadLog.java:299)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog.log(OWriteAheadLog.java:802)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.endAtomicOperation(OAtomicOperationsManager.java:82)
        at com.orientechnologies.orient.core.storage.impl.local.OStorageLocalAbstract.rollbackStorageTx(OStorageLocalAbstract.java:144)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.rollback(OLocalPaginatedStorage.java:1152)

###############################
java.io.IOException: Can not delete file. Retry limit exceeded.
        at com.orientechnologies.orient.core.storage.fs.OAbstractFile.delete(OAbstractFile.java:317) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.hashindex.local.cache.OWOWCache.doDeleteFile(OWOWCache.java:1039) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.hashindex.local.cache.OWOWCache.deleteFile(OWOWCache.java:704) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.hashindex.local.cache.OReadWriteDiskCache.deleteFile(OReadWriteDiskCache.java:300) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexAbstract.removeValuesContainer(OIndexAbstract.java:537) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexAbstract.rebuild(OIndexAbstract.java:368) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexAbstract.rebuild(OIndexAbstract.java:309) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexAbstract.addCluster(OIndexAbstract.java:597) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexAbstract.addCluster(OIndexAbstract.java:66) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexManagerAbstract.addClusterToIndex(OIndexManagerAbstract.java:170) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.index.OIndexManagerProxy.addClusterToIndex(OIndexManagerProxy.java:153) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.metadata.schema.OSchemaShared.createClassInternal(OSchemaShared.java:449) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.metadata.schema.OSchemaShared.doCreateClass(OSchemaShared.java:381) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.metadata.schema.OSchemaShared.createClass(OSchemaShared.java:314) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.metadata.schema.OSchemaProxy.createClass(OSchemaProxy.java:77) [com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
       

java.io.IOException: Can not delete file. Retry limit exceeded. (11).
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog$LogSegment.delete(OWriteAheadLog.java:299) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog.log(OWriteAheadLog.java:802) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWriteAheadLog.logFullCheckpointStart(OWriteAheadLog.java:963) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.makeFullCheckpoint(OLocalPaginatedStorage.java:1353) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
Wrapped by: com.orientechnologies.orient.core.exception.OStorageException: Error during checkpoint creation for storage baasbox
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.makeFullCheckpoint(OLocalPaginatedStorage.java:1362) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.restoreIfNeeded(OLocalPaginatedStorage.java:1560) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.open(OLocalPaginatedStorage.java:252) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
Wrapped by: com.orientechnologies.orient.core.exception.OStorageException: Cannot open local storage 'db/baasbox' with mode=rw
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.open(OLocalPaginatedStorage.java:258) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]
        at com.orientechnologies.orient.core.db.raw.ODatabaseRaw.open(ODatabaseRaw.java:107) ~[com.orientechnologies.orientdb-core-1.7.10.jar:1.7.10]

**Comment by laa** [comment]
May be you will just use patched version of delete file ? Borrowed from 2.0. 
```
acquireWriteLock();
 try {
      close();
      if (osFile != null) {
	final FileSystem fileSystem = FileSystems.getDefault();
        final Path path = fileSystem.getPath(osFile.getAbsolutePath());

        Files.delete(path);
    }
 } finally {
      releaseWriteLock();
 }
```

**Comment by brucejohnson** [comment]
Thanks,
I made this change in OAbstractFile.java and OWriteAheadLog.java and it resolved the issue.

**Comment by laa** [comment]
ok I mark it as invalid because it is already fixed in 2.0 versions.
